test/unit/visitor/defuse_analyze.cpp

```diff
@@ -315,6 +315,86 @@ SCENARIO("Perform DefUse analysis on NMODL constructs") {
         }
     }
 
+    GIVEN("local variable usage in else block") {
+        std::string nmodl_text = R"(
+            NEURON {
+                RANGE tau
+            }
+
+            DERIVATIVE states {
+                IF (tau == 1) {
+                } ELSE {
+                    LOCAL tau
+                    tau = 1 + tau
+                }
+            }
+        )";
+
+        std::string expected_text =
+            R"({"DerivativeBlock":[{"CONDITIONAL_BLOCK":[{"IF":[{"name":"U"}]},{"ELSE":[{"name":"LU"},{"name":"LD"}]}]}]})";
+
+        THEN("Def-Use chains should return USE because global variables have precedence in eval") {
+            std::string input = reindent_text(nmodl_text);
+            auto chains = run_defuse_visitor(input, "tau");
+            REQUIRE(chains[0].to_string() == expected_text);
+            REQUIRE(chains[0].eval() == DUState::U);
+        }
+    }
+
+    GIVEN("local variable conditional definition") {
+        std::string nmodl_text = R"(
+            NEURON {
+                RANGE beta
+            }
+
+            DERIVATIVE states {
+                LOCAL tau
+                IF (beta == 0) {
+                    tau = 1
+                }
+            }
+        )";
+
+        std::string expected_text =
+            R"({"DerivativeBlock":[{"CONDITIONAL_BLOCK":[{"IF":[{"name":"LD"}]}]}]})";
+
+        THEN("Def-Use chains should return USE because global variables have precedence in eval") {
+            std::string input = reindent_text(nmodl_text);
+            auto chains = run_defuse_visitor(input, "tau");
+            REQUIRE(chains[0].to_string() == expected_text);
+            REQUIRE(chains[0].eval() == DUState::CD);
+        }
+    }
+
+    GIVEN("local and range variables usage and definitions") {
+        std::string nmodl_text = R"(
+            NEURON {
+                RANGE beta
+            }
+
+            DERIVATIVE states {
+                LOCAL tau
+                IF (beta == 0) {
+                    tau = 1
+                } ELSE {
+                    beta = 0
+                }
+            }
+        )";
+
+        std::string expected_text =
+            R"({"DerivativeBlock":[{"CONDITIONAL_BLOCK":[{"name":"IF"},{"name":"ELSE"}]}]})";
+
+        THEN(
+            "Def-Use chains should return NONE because the variable we look for is not one of "
+            "them") {
+            std::string input = reindent_text(nmodl_text);
+            auto chains = run_defuse_visitor(input, "alpha");
+            REQUIRE(chains[0].to_string() == expected_text);
+            REQUIRE(chains[0].eval() == DUState::NONE);
+        }
+    }
+
```

Comment:
a few questions:
- can you declare also GLOBAL inside the functor?
- may we have also the following tests:
    - 2 local variables (used)
    - 2 local variables (1 unused)
    - 2 unused local variables
    - no local variables
    - a local variable in an IF ELSE  block? Is it possible?

Reply:
I added a test that includes some of these cases that were not included before. I think it gives a good indication of how the `DefUseAnalyzeVisitor` works. Let me know if you want to add more cases :+1: 
`GLOBAL` variables are defined only in the `NEURON` block